### PR TITLE
[23.1] Fix history items repeated search bug

### PR DESCRIPTION
**Pull request by ahmedhamidawan**
Fixes https://github.com/galaxyproject/galaxy/issues/16211

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: client/src/components/History/CurrentHistory/HistoryFilters/HistoryFilters.vue [hunk 1]
```diff
@@ -197,6 +197,12 @@ export default {
             this.onToggle();
             this.filterSettings["create_time>"] = this.create_time_gt;
             this.filterSettings["create_time<"] = this.create_time_lt;
+            if (this.filterSettings["deleted:"] == undefined) {
```
Comment (mvdbeek) [inline client/src/components/History/CurrentHistory/HistoryFilters/HistoryFilters.vue]: Can we have a test for this, should it use `===`, and under which circumstances is `this.filterSettings["deleted:"]` undefined ? A comment may help future us here.

Comment (ahmedhamidawan) [reply client/src/components/History/CurrentHistory/HistoryFilters/HistoryFilters.vue]: The `filterSettings` object is cleared when `deleted` or `visible` are set to `any`. This is what causes this bug, unless we explicitly reset those values when a search is performed again. Adding a comment here for that for now will help so will do that.
Other than that, working on another PR where I am trying to create a reusable `FilterMenu` component that will fix these bugs.

Comment (mvdbeek) [reply client/src/components/History/CurrentHistory/HistoryFilters/HistoryFilters.vue]: Sounds good, thank you!

**Review by mvdbeek — APPROVED** [review]
Thank you!

**Pull request merged**